authorization-server/src/modules/auth/strategies/jwt.strategy.ts

```diff
@@ -1,22 +1,34 @@
-import { ExtractJwt, Strategy } from 'passport-jwt';
+import { ExtractJwt, JwtFromRequestFunction, Strategy } from 'passport-jwt';
 import { PassportStrategy } from '@nestjs/passport';
 import { Injectable } from '@nestjs/common';
 import { ConfigService } from '@nestjs/config';
+import * as express from 'express';
 
 @Injectable()
 export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
   constructor(config: ConfigService) {
+    const extractorFunctions: JwtFromRequestFunction[] = [];
+
+    const fromHeader = ExtractJwt.fromAuthHeaderAsBearerToken();
+
+    const fromCookie = (req: express.Request) =>
+      req &&
+      req.cookies &&
+      req.cookies[config.get<string>('AUTH_COOKIE_NAME_ACCESS_TOKEN')];
+
+    if (config.get<boolean>('AUTH_COOKIE_ENABLED')) {
+      if (config.get<boolean>('AUTH_COOKIE_ONLY')) {
+        extractorFunctions.push(fromCookie);
+      } else {
+        extractorFunctions.push(fromHeader);
+        extractorFunctions.push(fromCookie);
+      }
+    } else {
+      extractorFunctions.push(fromHeader);
+    }
```

Comment:
I think this can also be simplified a bit (at the cost of negative logic in the first `if`)
```suggestion
    if (!config.get<boolean>('AUTH_COOKIE_ONLY')) {
      extractorFunctions.push(fromHeader);
    }
    if (config.get<boolean>('AUTH_COOKIE_ENABLED')) {
      extractorFunctions.push(fromCookie);
    }
```

Reply:
it needs to be like this when having the option of tokens in both cookies and bodies

Reply:
> it needs to be like this when having the option of tokens in both cookies and bodies

Could you explain this a bit more? The case you are describing is if the config is `AUTH_COOKIE_ONLY == false` && `AUTH_COOKIE_ENABLED == false`, right? In this case, wouldn't the suggested code populate `extractorFunction` with both `fromHeader` and `fromCookie` (which is what we want)?

Reply:
OK. Now I am getting what you mean. I am reducing this spaghetti.

Reply:
`AUTH_COOKIE_ONLY` can be set to true when `AUTH_COOKIE_ENABLED` is `false`. That does not make sense but is achievable by inpatient users. There is no validation for this, so my "ifs" cover this case.